service/history/task/cross_cluster_task_processor.go

```diff
@@ -222,7 +228,12 @@ func (p *crossClusterTaskProcessor) processTaskRequests(
 			crossClusterTask, future := NewCrossClusterTaskForTargetCluster(
 				p.shard,
 				taskRequest,
+				p.taskExecutor,
 				p.logger,
+				func(t Task) {
+					// the only possible error is processor shutdown which is safe to discard
```

Comment:
looks like there are some other errors but they were logged in processor_base.go? Even if it's a different error, we don't want to change how this function behaves right?

Reply:
Yes, `p.submitTask()` only return error during shutdown. Other errors will be logged and the task will be push to a re-dispatch queue for retrying the submission. Here we just need to make sure the `submitTask` is invoked so that task is either submitted or pushed to re-dispatch queue.